nightfall-optimist/src/classes/block.mjs

```diff
@@ -124,16 +121,19 @@ class Block {
     // note that the transactionHashes array is not part of the on-chain block
     // but we compute it here for convenience. It needs removing before sending
     // a block object to the blockchain.
-    return new Block({
-      proposer,
-      transactionHashes: transactions.map(t => t.transactionHash),
-      leafCount: timber.leafCount,
-      root: updatedTimber.root,
-      blockHash,
-      nCommitments,
-      blockNumberL2,
-      previousBlockHash,
-    });
+    return {
+      block: new Block({
```

Comment:
I return the `block` because now there is another parameter `updatedTimber` that is the updated timber in order to generate subsequent block proposed transactions until the unprocessed txs, that is needed in the `makeBlock` function that does the `Block.build` for every block.